### ISSUE TITLE
🥼🤡 Research content reorganisation

### ISSUE THREAD

**Issue by MistTheBear** [jenesis360/personal_website#17]
## 🪣 Contents of current `/research` page

- Themes
  - Susy and dark matter (why is this first, pretty sure as of 2023, this ain't a primary focus)
  - ATLAS and silicon detectors
  - Colliding light at the LHC
  - Axions and dark photons
  - (smaller) Origins of mass
  - (smaller) Theoretical physics (bit vague, no?)
  - (smaller) Cosmic ray science
- Papers (Publications?) - this looks comically short compared to the talks...
  - Phenomenology
  - Analysis
  - Instrumentation
  - Thesis
- Talks
  - Invited seminars
  - Conferences & workshop presentations
  - Collaboation plenaries
- Hardware (seems kinda random this has a separate top-level section?)
  - Silicon detectors, ATLAS inner tracker (ITk) strips upgrade, tabletop spectrometer
  - Informally structured at the moment
- Analysis (curiously missing the section header call "Analysis")
  - Photon collider science
  - Supersymmetry and dark matter

### How papers/publications are shown

- Title
- Contributors
- Links to publication(s)

### How talks are shown

- Invited seminars
  - Institution
  - Event description
  - Date
- Conferences & workshop presentations
  - Event name
  - Date
  - Title of presentation
- Collaboration plenaries
  - Event name
  - Date
  - Title of presentation

Interestingly, no abstract for any! Guess nobody got time for that!

## 🗃️ Reorganisation Options 

Theses are non-exclusive, choosing one does not exclude another from being adopted

### Thematic

- Group work output by theme
- Some work wil likely appear in multiple "themes"

### CV style

- Top level navigation by type of research output:
  - Publications
  - Talks

### Active and/or most recent work

- What are you currently working on or promoting?
- Bit of a blend between a "theme" and a "project" highlight

### Select highlights

- Same as active/most-recent but is time-insensitive
  - Could combine by attaching some indicator of "freshness" to any project being promoted/highlighted
- The hardware and analysis sections are both examples of theses
- Also get the opportunity to highlight specific talks (e.g. part of promotional tours for new publication)
  - All the effort spent on designing slides...may as well maximise recycling ♻️

### Top level navigation 

Have a curated section, and then the "CV dump" that people are more interested in

- 🌟 Recents/Highlights
  - 🐻 Curation is manual
  - 🤖 But can leverage automation, e.g. query by tag/category/theme/project blah blah
- 📔 Publications
  - 🤖 Just an auto-generated page
- 🦜 Talks
  - 🤖 Just an auto-generated page

### Project/Theme navigation

- Related to the curated recents/highlights where each "section" will be either a theme or a project
- Want the density of the publications/talks list, but have room for elucidation seen in the theme/project abstracts
- Ways this can be achieved?
  - Collapsible sections on the same page
  - Multi pages with a contents list/browser
- Note that the way content will be edited (see #7 ) will likely be identical regardless of design, since the content is separated from the GUI

**Comment by MistTheBear** [comment]
## Pages and Routes

- Homepage
- Me/CV
- Research
  - Individual research themes page are considered "nice to have", so can be omitted on initial release
- Beyond (basically all the other fluff)
  - This is considered "low priority", so can be omitted on initial release

Description | Route
-- | --
Homepage | `/`
me | `/me` or '/cv'
Research | `/research`
Research Themes | `/research/themes`
Research Publications | `/research/papers`
Research Talks | `/research/talks`
Research Theme Focus | `/research/themes/[themeName]`
Beyond | `/beyond`
Outreach | `/outreach` or `/beyond/outreach`
Design | `/design` or `/beyond/design`


### What goes in the primary navigation?

- Home
- Me/CV
- Research (dropdown for papers and talks?)
- [Later] Beyond (dropdown for outreach and design)

### What goes on the `/research` and `/beyond` hub pages

- Only the "highlights" or more recent examples
- Nudges to "See more" which take you into the deeper level pages

**Comment by MistTheBear** [comment]
## Page Content

### Home

- Brief bio (kinda what is on the home page right now)
  - Name
  - "Job" description
  - Institutional affiliation
  - Main research themes
  - Photo of you (cos this seems common in academic websites)
- Contact information
  - i.e. a link to your uni's profile page rather than actual plain text email
- Link to pdf CV
- Research highlights (equivalent to bits from `/research`

### Me/CV

- This is just the web form of your CV (doesn't actually exist on current website)
- Given that your research is in the research sections, the only real valuable information to be found here is the educational/work experience (i.e. what's in your "bio" at the moment)

### Research

- This page is intended to give an up-to-date summary of the latest research being done
  - Given it is curated, this can decay if not maintained
- If no curation given, it will just use defaults
  - 2 most recents talks
  - 2 most recent papers
  - themes mentioned in the 2 most recent talks/papers
- Borrows content from themes, papers, and talks
- Uses the curated highlights from paper and talks (probably up to 2)
- And uses a curated highlight of themes

#### Note on getting 2 most recent

- Astro's [`getCollection()`](https://docs.astro.build/en/reference/api-reference/#getcollection) does not support sorting
- Sorting is relatively trivial via the built-in JS `sort()` method
  - Most recent means descending (reverse-chronological) order
- Then to get the first 2, just use the built-in JS `slice()` method
  - Unlike `sort()` which is "in-place", `slice()` makes a shallow copy of the array, so need to assign it to a new variable
  - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice
- Also note the `const` in JS is a bit weird for arrays and objects which can still be updated!
  - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/const

```js
const allTalks = await getCollection('talks');
allTalks.sort(a, b => b.data.date - a.data.date);
const mostRecent2Talks = allTalks.slice(0, 2);
```

- Could conceivably use a helper method for this - which can be unit tested

### Themes

- Derived from the current research page: themes section
- Each theme section will contain
  - Name
  - Image
  - Description (try to be less verbose than you are now, it looks suspiciously like you've written each description so they have the same number of rendered lines lol)
  - Featured works (these are either papers or talks)
- The page will by default use alphabetical order of ALL themes
- But can be overriden by a custom import of the themes where you
  - Decide which order they appear in
  - Decide which ones appear at all (ones not manually selected are omitted)

### Papers/Talks

- Curated highlights (maximum of 2? or do we need more? Like surely you're not that proflific lol)
  - If abstract/description/content exists, it will be shown in the curated highlights
- List of all items (including the curated highlights btw!) in reverse chronological order
  - Basically a table on larger screens or a stack of mini cards on narrower screens
  - This is effectively a CV dump

**Comment by MistTheBear** [comment]
# Design and Layout

Proposed changes and new rules/guidelines relating to the appearance and functional layout of the website

## Guidelines and Motifs

### Principles

#### Similar _feel_ across narrow and wide screens

- Note that this is about how the website _feels_
- There are naturally going to be differences on narrow and wide screens
- But the objective is to minimise the perceived gap between narrow and wide screen experiences
- Keep navigation as similar as possible across screen sizes
  - Avoid hamburger collapsed menus
  - Would rather have a nav tab that persists across all screen sizes
- Provide some form of intra-page navigation that persists at all scroll levels
  - Rather than jump navs that are at the top of the page (like now) and then relying on a "jump to top" experience

#### No horizontal scrolling 

- Horizontal scrolling should not be a solution to solve the narrow screen elongation problem
- There is no good input method for horizontal scrolling
  - Trackpad? You're competing wrist pivoting (horizontal) with hand-finger extension and retraction (vertical scrolling)
  - Scroll wheel? Most mice only have one scroll wheel which is single axis. If you're using fancier mice with an extra scroll wheel...well...how often do you even use that?
  - Touchscreen? The ever reliable (or not) experience of scrolling sideways...I'm so lazy, that the cognitive effort for 2D scrolling is beyond me
- Diffculty with indicating horizontal scrollability
- Apple's current store is a case in point of crap horizontal scrolling experience

#### Colour Accenting

- Colour for accenting only
  - This isn't to say minimise use of colour
  - But more to impart greater meaning to the use of colour (attention drawing)
- Don't use colour (chroma) alone as a differentiation method
  - Contrast matters
  - Use visual patterns on top of colour differentiation if required
- Use at most 2 accent colours, and ideally just the one
- Consider just using black & white (or softened equivalents like dark-grey & off-white) in v1
- Then start adding more colours in v2

#### CSS over JS

- Where possible, use CSS rather than JS
- This can potentially mean using cutting edge CSS rather than older JS workarounds
  - Evergreen browsers eventually update
  - JS is more code to maintain
- Also an excuse to keep the design flourishes simple
  - If it's a pain in the arse to do in CSS, then don't do it!
- JS should ideally be used to enhance something functional

## Layouts

- These are Astro layouts
- Individual pages will use these layouts and pass props to the layouts

### Homepage

- Likely to be the only unique page layout on the site
- Keep things simple by having standard 1D layout (multiple rows) with nesting (columns within those rows) in v1
- Can experiment with 2D layouts with CSS Grid on larger screens in a v2

#### 1D

- More conventional row-column structure using either Flexbox or Grid
- Effectively grouping related content into the same row

```html
<main>
    <section id="intro">
        <section id="bio"></section>
        <div id="contact"></div >
    </section>
    <section id="research">
        <header></header>
        <section id="themes"></section>
        <section id="papers"></section>
        <section id="talks"></section>
    </section>
    <section id="beyond"></section>
</main>
<footer></footer>
```

#### 2D

- Leverage CSS Grid to position elements in a 2D grid
- More freeform doc with each top-level section being treated equally

```html
<main>
    <section id="bio"></section>
    <section id="themes">
        <header></header>
        <article></article>
        <article></article>
    </section>
    <section id="papers">
        <header></header>
        <article></article>
        <article></article>
    </section>
    <section id="talks"></section>
    <section id="beyond"></section>
</main>
<footer></footer>
```

### CV

- Effectively a higher density combine collections page
- May need a new collection for "institutions" which covers both education and working instutitions
- Only really unique thing about it is the initial content which can be the equivalent of a statement plus a link to a PDF version of the CV

### Collections

- Applies primarily to the papers and talks pages
- But realistically works for any other collection
